domain/lib/src/model/sharedspace/shared_space_node_nested.dart

```diff
@@ -0,0 +1,62 @@
+// LinShare is an open source filesharing software, part of the LinPKI software
+// suite, developed by Linagora.
+//
+// Copyright (C) 2020 LINAGORA
+//
+// This program is free software: you can redistribute it and/or modify it under the
+// terms of the GNU Affero General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later version,
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:equatable/equatable.dart';
+import 'package:domain/domain.dart';
+
+class SharedSpaceNodeNested extends Equatable {
```

Comment:
[Confirm] I don't know why the class name has `NodeNested` at the end @@. I think it just `ShareSpace`. Because Ive checked a api that query all node in side of share space, a response like:
```
[
  {
    "type": "DOCUMENT",
    "uuid": "fa563e82-ff7b-434a-9169-2e174c258bdb",
    "name": "java_error_in_STUDIO_28915.log",
    "parent": "18309bb8-2ed2-4782-b6b8-7542917db348",
    "workGroup": "e352ed55-abef-4630-816f-c025caa3b025",
    "description": null,
    "metaData": null,
    "lastAuthor": {
      "name": "John Doe",
      "mail": "user1@linshare.org",
      "uuid": "1f75190b-73aa-4c4b-9efb-2760be07c3bb",
      "accountType": "INTERNAL"
    },
    "creationDate": 1607593227197,
    "modificationDate": 1607593227197,
    "size": 6039,
    "mimeType": "text/plain",
    "hasRevision": false,
    "sha256sum": null,
    "uploadDate": 1607593227197,
    "hasThumbnail": null
  }
]
```
I think the name `SharedSpaceNodeNested` more appropriate for it.

Reply:
I cloned the name from Backend, we have 2 apis relate to `SharedSpace` ans `SharedSpaceNodeNested`, please check API on Android for more details